pkg/agent/proxy/proxier.go

```diff
@@ -1010,6 +1015,54 @@ func (p *proxier) GetServiceFlowKeys(serviceName, namespace string) ([]string, [
 	return flows, groups, found
 }
 
+func (p *proxier) HandlePacketIn(pktIn *ofctrl.PacketIn) error {
+	if pktIn == nil {
+		return errors.New("empty packetin for Antrea Proxy")
+	}
```

Comment:
You should check if the packetIn message is for Service no Endpoint purpose first. Otherwise, all packet_in message with reason "PacketInReasonSvcReject|PacketInReasonNP|PacketInReasonMC" would be processed by the later logic because they three are sharing the same value.

Reply:
Thanks, updated.

Reply:
where is it added?

Reply:
reminder